### PR TITLE
Add Python 3.12 support

### LINKED ISSUE #80: Add Python 3.12 support
Releasing on 2 October https://docs.python.org/3.12/whatsnew/3.12.html

### PR DESCRIPTION
**Pull request by jashparekh**
Closes #80

### CODE REVIEW

**Comment by jashparekh** [comment]
> @jashparekh Hello and thank you 🙏 for the PR. I did not realize CI would only run with my permission till now.
> 
> Your changes look good, but there are some errors when running tests:
> 
> ```
> ERROR src/redflag/__init__.py - ModuleNotFoundError: No module named 'pkg_resources'
> ERROR src/redflag/_version.py - ModuleNotFoundError: No module named 'pkg_resources'
> ERROR tests/test_sklearn.py - ModuleNotFoundError: No module named 'pkg_resources'
> ```
> 
> I have not tried to solve this and don't have any suggestions I'm afraid, but maybe you can take a look if you have time.
> 
> Cheers!

Thanks, can you approve my request to run the CI pipeline

File: pyproject.toml [hunk 1]
```diff
@@ -32,6 +33,7 @@ dependencies = [
 
 [project.optional-dependencies]
 dev = [
+    "setuptools"
```
Comment (kwinkunks) [inline pyproject.toml:36]: You needa comma at the end of the line.

Comment (jashparekh) [reply pyproject.toml]: Thank you, just added it

**Comment by kwinkunks** [comment]
This is wonderful stuff @jashparekh -- thank you for persevering, and thank you for the PR 🙌

**Review by kwinkunks — APPROVED** [review]
LGTM

**Pull request merged**